pkg/kubelet/kubelet.go

```diff
@@ -387,6 +387,14 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		}
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.DisableCloudProviders) && cloudprovider.IsDeprecatedInternal(cloudProvider) {
+		err := fmt.Errorf(
+			"cloud provider %q was specified, but built-in cloud providers are disabled. Please set --cloud-provider=external and migrate to an external cloud provider",
+			cloudProvider)
+		err = fmt.Errorf("%s: You might need to use out-of-tree credential providers. Reach to sig-cloud-provider for guidance", err)
```

Comment:
we can actually remove this for now, because out-of-tree credentials providers run independent of what is set for `--cloud-provider`

Reply:
also nit: no need to define the error on line 391, just return it inline on line n395